Framework/Algorithms/src/SumEventsByLogValue.cpp

```diff
@@ -203,33 +207,28 @@ void SumEventsByLogValue::createTableOutput(const Kernel::TimeSeriesProperty<int
   for (int value = minVal; value <= maxVal; ++value) {
     const auto row = std::size_t(value - minVal);
     // Create a filter giving the times when this log has the current value
-    SplittingIntervalVec filter;
-    log->makeFilterByValue(filter, value, value); // min & max are the same of course
 
+    TimeROI timeRoi;
+    const TimeROI *temp = &m_inputWorkspace->run().getTimeROI();
     // This section ensures that the filter goes to the end of the run
     if (value == log->lastValue() && protonChargeLog) {
-      TimeInterval timeAfterLastLogValue(log->lastTime(), m_inputWorkspace->getLastPulseTime());
-      log->expandFilterToRange(filter, value, value, timeAfterLastLogValue);
-    }
-
-    // convert to the actual time filter
-    TimeROI timeROI; // TODO remove this conversion
-    for (const auto &interval : filter) {
-      if (interval.start() < interval.stop())
-        timeROI.addROI(interval.start(), interval.stop());
+      const TimeInterval timeAfterLastLogValue(log->lastTime(), m_inputWorkspace->getLastPulseTime());
+      timeRoi = log->makeFilterByValue(value, value, true, timeAfterLastLogValue, 0.0, false, temp);
+    } else {
+      timeRoi = log->makeFilterByValue(value, value, false, TimeInterval(0, 1), 0., false, temp);
     }
 
     // Calculate the time covered by this log value and add it to the table
-    double duration = 0.0;
-    for (auto &time : filter) {
-      duration += time.duration();
-    }
-    timeCol->cell<double>(row) = duration;
+    Run run(m_inputWorkspace->run());
+    run.setTimeROI(timeRoi);
+    timeCol->cell<double>(row) = run.getTimeROI().durationInSeconds();
 
     interruption_point();
     // Sum up the proton charge for this log value
-    if (protonChargeLog)
-      protonChgCol->cell<double>(row) = sumProtonCharge(protonChargeLog, timeROI);
+    if (protonChargeLog) {
+      const double currentConversion = 1.e-6 / 3600.;
```

Comment:
Please add a comment as to why this conversion factor s here and what it converts between.